compiler/src/dmd/tokens.d

```diff
@@ -900,7 +900,7 @@ extern (C++) struct Token
 
 nothrow:
 
-    int isKeyword() const @safe
+    int isKeyword() pure const @safe @nogc
     {
         foreach (kw; keywords)
```

Comment:
Aside from review of this PR, this seems ultra inefficient code for lookup.

Reply:
Good point, do you think `std.algorithm.searching.canFind` is any better? I'm guessing it's at least more optimal than a linear search

Reply:
The ideal way would be to use a hash set, but it seems like Phobos does not offer that container as of now. A red-black tree seems overkill (and also overly cumbersome) for this. I tried using an AA with `bool[TOK]`, but it cannot be statically initialized :(

Reply:
canFind does the same linear search, and dmd's source code may not use Phobos. The PR was fine as it was, if you want to optimize it please do so in a separate PR (though I don't think it's needed at the moment).

Reply:
Righto, reverting it

Reply:
Yeah, just a side note, this should be addressed in a follow up, indeed.

Reply:
By reading the code, maybe smart compilers might just generate jump tables here, but would be better to transform this to a `static foreach` + `switch case` or even better, a bitarray, which would always be the fastest optimization and O(1).